ibm_mq/datadog_checks/ibm_mq/config.py

```diff
@@ -99,6 +99,12 @@ def __init__(self, instance):
         self._queue_tag_re = instance.get('queue_tag_re', {})
         self.queue_tag_re = self._compile_tag_re()
 
+        try:
+            mqcd_version = instance.get('mqcd_version', 6)
+            self.mqcd_version = getattr(pymqi.CMQC, 'MQCD_VERSION_{}'.format(int(mqcd_version)))
+        except (ValueError, AttributeError):
+            raise ConfigurationError("mqcd_version must be a number between 1 and 9. {} found.".format(mqcd_version))
```

Comment:
Suggested refactoring to deal with the two possible errors (not an int/not a supported version) separately:

```suggestion
        raw_mqcd_version = instance.get('mqcd_version', 6)

        try:
            mqcd_version = int(raw_mqcd_version)
        except (TypeError, ValueError):
            raise ConfigurationError("mqcd_version must be convertible to an integer, got {!r}".format(raw_mqcd_version))

        try:
            self.mqcd_version = getattr(pymqi.CMQC, 'MQCD_VERSION_{}'.format(mcqd_version))
        except AttributeError:
            raise ConfigurationError("Only MQCD versions 1 through are supported, got {}.".format(mqcd_version))
```

Reply:
I wanted to avoid the situation where someone puts for example 'MQCD_VERSION_10' they get an error saying that it must be an int then they put 10 and get a different error saying that has to be max 9 and give them all the information they may need

Reply:
I see, okay. Then maybe this, to resolve the ambiguity that prompted Alex's comment?

```suggestion
        raw_mqcd_version = instance.get('mqcd_version', 6)
        try:
            mqcd_version = int(raw_mqcd_version)
            self.mqcd_version = getattr(pymqi.CMQC, 'MQCD_VERSION_{}'.format(mcqd_version))
        except (ValueError, TypeError, AttributeError):
            raise ConfigurationError("mqcd_version must be a number between 1 and 9, was {!r}.".format(raw_mqcd_version))
```

Reply:
I'm ok with renaming to `raw_mqcd_version` see why we need an int `mqcd_version` too. In fact I'm thinking we don't need to cast to int at all

Reply:
> we don't need to cast to int at all

Ah, indeed 🤦‍♂ that should do it!